samples/sample_encode/src/pipeline_encode.cpp

```diff
@@ -2009,18 +2014,29 @@ void CEncodingPipeline::FreeFileWriters()
 
 mfxStatus CEncodingPipeline::FillBuffers()
 {
+    mfxStatus sts = MFX_ERR_NONE;
+
     if (m_nPerfOpt)
     {
         for (mfxU32 i = 0; i < m_nPerfOpt; i++)
         {
             mfxFrameSurface1* surface = m_pmfxVPP ? &m_pVppSurfaces[i] : &m_pEncSurfaces[i];
-
-            mfxStatus sts = m_pMFXAllocator->Lock(m_pMFXAllocator->pthis, surface->Data.MemId, &surface->Data);
-            MSDK_CHECK_STATUS(sts, "m_pMFXAllocator->Lock failed");
-            sts = m_FileReader.LoadNextFrame(surface);
-            MSDK_CHECK_STATUS(sts, "m_FileReader.LoadNextFrame failed");
-            sts = m_pMFXAllocator->Unlock(m_pMFXAllocator->pthis, surface->Data.MemId, &surface->Data);
-            MSDK_CHECK_STATUS(sts, "m_pMFXAllocator->Unlock failed");
+#if defined(LIBVA_DRM_SUPPORT)
+            if (m_nVACopy != -1) // vacopy mode
+            {
+                sts = LoadFrameWithVACopy(surface, m_nVACopy);
```

Comment:
What if X11 backend was selected?

Reply:
This PR only implemented for DRM backend according to customer's requirement. For other backends, this option will be disabled by default. If we need to support X11 backend, we may add vaCopy related functions to CVAAPIDeviceX11 in future.